libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/TxGen.hs

```diff
@@ -199,6 +205,50 @@ lookupScript scriptHash mTag = do
           Just . snd <$> lookupByKeyM "plutusScript" (scriptHash, tag) gsPlutusScripts
     _ -> pure Nothing
 
+-- =====================================
+
+genGenericScriptWitness ::
+  (Reflect era) =>
+  Proof era ->
+  Maybe Tag ->
+  Script era ->
+  GenRS era (SafeHash (EraCrypto era) EraIndependentTxBody -> [WitnessesField era])
+genGenericScriptWitness proof mTag script =
+  case proof of
+    Shelley _ -> mkMultiSigWit proof mTag script
+    Allegra _ -> mkTimelockWit proof mTag script
+    Mary _ -> mkTimelockWit proof mTag script
+    Alonzo _ -> case script of
+      (TimelockScript timelock) -> mkTimelockWit proof mTag timelock
+      (PlutusScript _ _) -> pure (const [])
+    Babbage _ -> case script of
+      (TimelockScript timelock) -> mkTimelockWit proof mTag timelock
+      (PlutusScript _ _) -> pure (const [])
+    Conway _ -> case script of
+      (TimelockScript timelock) -> mkTimelockWit proof mTag timelock
+      (PlutusScript _ _) -> pure (const [])
```

Comment:
Love the switch to nested case statements. 

Slight improvement, redundant parenthesis:
```suggestion
      TimelockScript timelock -> mkTimelockWit proof mTag timelock
      PlutusScript _ _ -> pure (const [])
    Babbage _ -> case script of
      TimelockScript timelock -> mkTimelockWit proof mTag timelock
      PlutusScript _ _ -> pure (const [])
    Conway _ -> case script of
      TimelockScript timelock -> mkTimelockWit proof mTag timelock
      PlutusScript _ _ -> pure (const [])
```

Reply:
Changing to nested case statement had a small but not insignificant improvement in compile time. Makes it clearer too.